### PR TITLE
test: await focusing username element

### LINKED ISSUE #3447: test runner shuts down on auth test failure
https://github.com/determined-ai/determined/issues/3444

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- adds a missing await for taiko.focus which introduced an unexpected behavior

## Commentary (optional)

the sign in test passes without focusing as the username element should be in focus on page load to begin with

fixes #3447 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hamidzr** [comment]
I have some other changes to the auth test suite that I separated out to reduce the scope for this one https://github.com/determined-ai/determined/pull/3443

File: webui/tests/tests/StepImplementation.ts [hunk 1]
```diff
@@ -149,7 +149,9 @@ export default class StepImplementation {
   @Step('Sign in as <username> with <password>')
   public async signInWithPassword(username: string, password: string) {
     await goto(`${BASE_URL}/login`);
-    await t.clear(t.focus(t.$('#login_username')));
+    const loginSelector = t.$('#login_username');
+    await t.focus(loginSelector);
+    await t.clear(loginSelector);
```
Comment (hkang1) [inline webui/tests/tests/StepImplementation.ts:154]: can confirm that this section doesn't crash when the login input is not visible:
```
# Authentication
  ## Sign out	 ✔
  ## Sign in as user without password	 ✘
        Failed Step: Sign in as "user-wo-pw" without password
        Specification: specs/02-authentication.spec:12
        Error Message: CustomSelector with query #login_username is not visible
        Stacktrace: 
        Error: CustomSelector with query #login_username is not visible
            at /Users/calebkang/Projects/determined/webui/tests/node_modules/taiko/lib/actions/pageActionChecks.js:189:11
            at waitAndGetActionableElement (/Users/calebkang/Projects/determined/webui/tests/node_modules/taiko/lib/actions/pageActionChecks.js:156:3)
            at focus (/Users/calebkang/Projects/determined/webui/tests/node_modules/taiko/lib/actions/focus.js:12:12)
            at Object.module.exports.focus (/Users/calebkang/Projects/determined/webui/tests/node_modules/taiko/lib/taiko.js:1038:16)
            at Object.module.exports.<computed> [as focus] (/Users/calebkang/Projects/determined/webui/tests/node_modules/taiko/lib/taiko.js:2612:14)
  ## Sign in as user with password	 ✔ ✔
  ## Sign in	 ✔
```

**Review by hkang1 — APPROVED** [review]
Can confirm that guage is no longer crashing when attempting to sign in.

Nice work with this!

File: webui/tests/tests/StepImplementation.ts (hunk 1 above)
Comment (hkang1) [inline webui/tests/tests/StepImplementation.ts:154]: **question:** once a focus has been achieved, we might be able to just call `await t.clear()` without the selector?

Comment (hamidzr) [reply webui/tests/tests/StepImplementation.ts:154]: yeah it seems so https://docs.taiko.dev/api/clear/ I'll update

Comment (hkang1) [reply webui/tests/tests/StepImplementation.ts]: **suggestion:** I'm trying this out currently on top of your PR and it's been pretty stable so far. I think it addresses the `visibility` issue. wdy think?
```
  @Step('Sign in as <username> with <password>')
  public async signInWithPassword(username: string, password: string) {
    const usernameId = 'login_username';
    const passwordId = 'login_password';
    await goto(`${BASE_URL}/login`);
    await t.textBox({ id: usernameId }).isVisible();
    await t.focus(t.textBox({ id: usernameId }));
    await t.clear();
    await t.write(username);
    if (password !== '') {
      await t.textBox({ id: passwordId }).isVisible();
      await t.focus(t.textBox({ id: passwordId }));
      await t.write(password);
    }
    await clickAndWaitForPage(t.button('Sign In'));
    await t.text(username, t.near(t.$('#avatar'))).exists();
  }
```

Comment (hamidzr) [reply webui/tests/tests/StepImplementation.ts]: do you wanna submit that as a separate pr?

File: webui/tests/tests/StepImplementation.ts [hunk 2]
```diff
@@ -149,7 +149,9 @@ export default class StepImplementation {
   @Step('Sign in as <username> with <password>')
   public async signInWithPassword(username: string, password: string) {
     await goto(`${BASE_URL}/login`);
-    await t.clear(t.focus(t.$('#login_username')));
+    const loginSelector = t.$('#login_username');
+    await t.focus(loginSelector);
```
Comment (hamidzr) [inline webui/tests/tests/StepImplementation.ts:153]: ah coulda been one line

**Pull request merged**